### PR TITLE
Support reusing the built image while submitting ElasticDL job

### PR DESCRIPTION
**Pull request by brightcoder01**

### CODE REVIEW

File: elasticdl/python/common/args.py [hunk 1]
```diff
@@ -107,7 +107,13 @@ def add_common_params(parser):
         help="The repository for generated Docker images, if set, the image "
         "is also pushed to the repository",
     )
-    parser.add_argument("--image_base", help="Base Docker image.")
+    parser.add_argument("--image_base", default="", help="Base Docker image.")
+    parser.add_argument(
+        "--image_name",
+        default="",
+        help="The image for this job. If set, "
```
Comment (skydoorkai) [inline elasticdl/python/common/args.py]: help="If set, use this image instead of building one.",

File: elasticdl/python/elasticdl/api.py [hunk 2]
```diff
@@ -31,39 +35,44 @@ def train(args):
...
-            build_arguments_from_parsed_result(
-                args, filter_args=["model_zoo", "cluster_spec", "worker_image"]
-            )
+    container_args = [
+        "-m",
+        "elasticdl.python.master.main",
+        "--worker_image",
+        image_name,
+        "--model_zoo",
+        _model_zoo_in_docker(model_zoo),
```
Comment (skydoorkai) [inline elasticdl/python/elasticdl/api.py]: If not building image, model_zoo should not set from `_model_zoo_in_docker`.

File: elasticdl/python/elasticdl/api.py [hunk 3]
```diff
@@ -31,39 +35,44 @@ def train(args):
...
+        "--cluster_spec",
+        _cluster_spec_def_in_docker(args.cluster_spec),
```
Comment (skydoorkai) [inline elasticdl/python/elasticdl/api.py:50]: Also, if not building image, `cluster_spec` should not set from `_cluster_spec_def_in_docker `.

File: elasticdl/python/common/args.py (hunk 1 above)
Comment (brightcoder01) [reply elasticdl/python/common/args.py]: Good suggestion, updated.

File: elasticdl/python/elasticdl/api.py (hunk 2 above)
Comment (brightcoder01) [reply elasticdl/python/elasticdl/api.py]: Good suggestion, updated.

File: elasticdl/python/elasticdl/api.py (hunk 3 above)
Comment (brightcoder01) [reply elasticdl/python/elasticdl/api.py:50]: Good suggestion, updated.

**Review by skydoorkai — APPROVED** [review]

**Comment by skydoorkai** [comment]
Also need to merge into branch-0.2.

**Pull request merged**